stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/channel/internal/DatabaseChannelRepository.kt

```diff
@@ -258,11 +210,24 @@ internal class DatabaseChannelRepository(
                 channel.apply {
                     lastMessageAt = messageCreatedAt
                     messages = listOf(lastMessage)
-                }.also { insertChannel(it) }
+                }.also { upsertChannel(it) }
             }
         }
     }
 
+    private suspend fun Channel.convertToEntity(): ChannelEntity {
+        val dbChannel = channelDao.select(this.cid)
+
+        val thisLastMessageAt = this.lastMessage?.createdAt ?: this.lastMessageAt ?: Date(0)
+        return if (dbChannel?.lastMessageAt?.after(thisLastMessageAt) == true) {
+            this.lastMessageAt = dbChannel.lastMessageAt
```

Comment:
Hey @leandroBorgesFerreira,

I'm curious if could move property value altering for `this` message off `Channel.convertToEntity` function, since `convert` doesn't imply it's gonna modify the source object.